src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -389,11 +389,11 @@ private void AddClassifiedSpansForTokens(IEditorClassificationService classifica
                     span.Snapshot.AsText(), span.Span.ToTextSpan(), classifiedSpans, CancellationToken.None);
             }
 
-            private void OnDocumentActiveContextChanged(object sender, DocumentEventArgs args)
+            private void OnDocumentActiveContextChanged(object sender, DocumentActiveContextChangedEventArgs args)
             {
-                if (_workspace != null)
+                if (_workspace != null && _workspace == args.Workspace)
                 {
-                    ParseIfThisDocument(null, args.Document.Project.Solution, args.Document.Id);
+                    ParseIfThisDocument(null, args.Workspace.CurrentSolution, args.NewContextDocumentId);
```

Comment:
These changes to use `CurrentSolution` are a *little* bit worrisome.  Previously we captured it before the call to `ScheduleTask` and now we don't capture it until the event is actually raised.

Are you satisfied that the change is acceptable?